lib/rucio/core/authentication.py

```diff
@@ -301,6 +302,26 @@ def get_ssh_challenge_token(account, appid, ip=None, session=None):
     return new_challenge_token
 
 
+@transactional_session
+def get_auth_token_saml(account, saml_nameid, appid, ip=None, session=None):
+
+    # Make sure the account exists
+    if not account_exists(account, session=session):
+        return None
+
+    # remove expired tokens
+    session.query(models.Token).filter(models.Token.expired_at < datetime.datetime.utcnow(),
```

Comment:
please add the `for_update` statement, like in https://github.com/rucio/rucio/pull/2721/files